tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-with-anon-func-exprs.bal

```diff
@@ -0,0 +1,489 @@
+    string v = f3();
+    assertEqual(v, "Alex George");
+
+    f1 = fn1[1];
+    f2 = f1();
+    f3 = f2();
+    v = f3();
+    assertEqual(v, "Ranjan Fonseka");
+
+    f1 = fn1[2];
+    f2 = f1();
+    f3 = f2();
+    v = f3();
+    assertEqual(v, "John David");
+
+    var fn2 = from int i in [1, 2, 3, 4]
+              let function () returns (function () returns (int)) func = function () returns
+                                                           (function () returns (int)) => foo
+              select function () returns int {
+                  function () returns (int) ff = func();
+                  int val = ff();
+                  return val + i;
+              };
+
+    assertEqual(4, fn2.length());
+    function () returns int f4 = fn2[0];
+    int sum = f4();
+    assertEqual(sum, 4);
+
+    f4 = fn2[1];
+    sum = f4();
+    assertEqual(sum, 5);
+
+    f4 = fn2[2];
+    sum = f4();
+    assertEqual(sum, 6);
+
+    f4 = fn2[3];
+    sum = f4();
+    assertEqual(sum, 7);
+
+    var fn3 = from var {firstName: fnm, lastName: lnm, age: a} in personList
+              select function () returns string {
+                  function (string, string) returns (string) anonFunction =
+                  function (string x, string y) returns (string) {
+                      return x + " " + y;
+                  };
+                  return anonFunction(fnm, lnm);
+             };
+
+    assertEqual(3, fn3.length());
+    function () returns string f5 = fn3[0];
+    string nm = f5();
+    assertEqual(nm, "Alex George");
+
+    f5 = fn3[1];
+    nm = f5();
+    assertEqual(nm, "Ranjan Fonseka");
+
+    f5 = fn3[2];
+    nm = f5();
+    assertEqual(nm, "John David");
+}
+
+function foo() returns int {
+    return 1 + 2;
+}
+
+int[] arr = [3, 4, 5];
+(function () returns int)[] globalFn1 = from int m in arr
+                                        select function () returns int => m * y;
+
+(function () returns string)[] globalFn2 = from Person p in personList
+                                           select function () returns string => p.firstName + " " + p.lastName;
+
+int[] globalFn3 = from int m in arr
+                  let function () returns int func = function () returns int => m * y
+                  where func === function () returns int {
+                      return 20;
+                  }
+                  select m;
+
+(function () returns string)[] globalFn4 = from var {firstName, lastName, age} in personList
+                                           where firstName == "Alex"
+                                           select function () returns string {
+                                               string hello = "Hello ";
+                                               return hello + firstName + " " + lastName;
+                                           };
+
+type Card record {|
+   string msg;
+   int noOfEmp;
+|};
+
+(function () returns Card)[] globalFn5 = from var person in personList
```

Comment:
Trying to use the argument within the arrow function crashed at runtime. We can handle it separately if needed.
```ballerina
(function () returns string)[] globalFn5 = ttttFunc("sdasd");

function ttttFunc(string abc) returns (function () returns string)[] {
    var res = from var person in personList
                                             from var empFunc in (from var {firstName, lastName, dept} in employeeList
                                             select function () returns string => abc + " " + dept)
                                             select empFunc;
    return res;
}
```

Reply:
It seems this is not specific to argument being used in an anonymous function in the query. Using an argument in an inner query in a nested query crashes at runtime. The following crashes at runtime.
```ballerina
function foo(string msg) returns string[] {
    var res = from var person in personList
              from var emp in (from var {firstName, lastName, dept} in employeeList
                               select msg + " " + firstName)
               select emp;
    return res;
}
```
Created https://github.com/ballerina-platform/ballerina-lang/issues/31629. @pcnfernando Shall we fix this separately?

Reply:
Tried with a `join` instead of a nested `from`. Couldn't reproduce. +1 let's handle it separately